eclair-core/src/test/scala/fr/acinq/eclair/integration/MessageIntegrationSpec.scala

```diff
@@ -142,6 +143,39 @@ class MessageIntegrationSpec extends IntegrationSpec {
     assert(r.finalPayload.records.unknown.toSet == Set(GenericTlv(UInt64(113), hex"010203"), GenericTlv(UInt64(117), hex"0102")))
   }
 
+  test("send very large message with hop") {
+    // Total message size stays below 65536 bytes, the message is relayed.
+    val bytes = randomBytes(65285)
+    val encodedBytes = genericTlv.encode(GenericTlv(UInt64(135), bytes)).require.bytes
+
+    val alice = new EclairImpl(nodes("A"))
+
+    val probe = TestProbe()
+    val eventListener = TestProbe()
+    nodes("C").system.eventStream.subscribe(eventListener.ref, classOf[OnionMessages.ReceiveMessage])
+    alice.sendOnionMessage(nodes("B").nodeParams.nodeId :: Nil, Left(nodes("C").nodeParams.nodeId), None, encodedBytes).pipeTo(probe.ref)
+    assert(probe.expectMsgType[SendOnionMessageResponse].sent)
+
+    val r = eventListener.expectMsgType[OnionMessages.ReceiveMessage](max = 60 seconds)
+    assert(r.finalPayload.records.unknown.toSet == Set(GenericTlv(UInt64(135), bytes)))
+  }
+
+  test("send too large message with hop") {
+    // Total message size goes above the 65536 bytes limit.
+    val bytes = randomBytes(65290)
+    val encodedBytes = genericTlv.encode(GenericTlv(UInt64(135), bytes)).require.bytes
+
+    val alice = new EclairImpl(nodes("A"))
+
+    val probe = TestProbe()
+    val eventListener = TestProbe()
+    nodes("C").system.eventStream.subscribe(eventListener.ref, classOf[OnionMessages.ReceiveMessage])
+    alice.sendOnionMessage(nodes("B").nodeParams.nodeId :: Nil, Left(nodes("C").nodeParams.nodeId), None, encodedBytes).pipeTo(probe.ref)
+    assert(probe.expectMsgType[SendOnionMessageResponse].sent)
```

Comment:
It's surprising that we actually send that message, shouldn't it overflow at creation time?
It feels like `alice.sendOnionMessage` should fail here without sending an outgoing message, shouldn't it?

Reply:
Changed it so that it fails loudly